pontoon/contributors/utils.py

```diff
@@ -478,39 +490,55 @@ def get_contribution_timeline_data(user, contribution_type=None, day=None):
-            )
-            contribution_data[key]["url"] = f"{url}?{urlencode(params)}"
-
-        # Generate title for the localizations belonging to the same contribution type
-        if contribution_type == "user_translations":
-            title = f"Submitted { intcomma(c_count) } translation{ pluralize(c_count) } in { intcomma(p_count) } project{ pluralize(p_count) }"
-        elif contribution_type == "user_reviews":
-            title = f"Reviewed { intcomma(c_count) } suggestion{ pluralize(c_count) } in { intcomma(p_count) } project{ pluralize(p_count) }"
-        elif contribution_type == "peer_reviews":
-            title = f"Received review for { intcomma(c_count) } suggestion{ pluralize(c_count) } in { intcomma(p_count) } project{ pluralize(p_count) }"
-
-        contributions.update(
-            {
-                title: {
-                    "data": contribution_data,
+        for month, data in contribution_data.items():
+            total_count = sum([info["count"] for _, info in data.items()])
+            p_count = len(data)
+
+            # Generate title for the localizations belonging to the same contribution type
+            if contribution_type == "user_translations":
+                title = f"Submitted { intcomma(total_count) } translation{ pluralize(total_count) } in { intcomma(p_count) } project{ pluralize(p_count) }"
+            elif contribution_type == "user_reviews":
+                title = f"Reviewed { intcomma(total_count) } suggestion{ pluralize(total_count) } in { intcomma(p_count) } project{ pluralize(p_count) }"
+            elif contribution_type == "peer_reviews":
+                title = f"Received review for { intcomma(total_count) } suggestion{ pluralize(total_count) } in { intcomma(p_count) } project{ pluralize(p_count) }"
+
+            data["title"] = title
+
+            # Generate localization URL and add it to the data dict
+            params = params_map[contribution_type]
+            for field, val in data.items():
+                if field != "title":
+                    url = reverse(
+                        "pontoon.translate",
+                        args=[
+                            val["locale"]["code"],
+                            val["project"]["slug"],
+                            "all-resources",
+                        ],
+                    )
+                    val["url"] = f"{url}?{urlencode(params)}"
+
+        for month, data in contribution_data.items():
+            if month not in contributions:
+                contributions[month] = {}
+
+            if contribution_type not in contributions[month]:
+                contributions[month][contribution_type] = {}
+
+            contributions[month][contribution_type].update(
+                {
+                    "data": data,
                     "type": contribution_type.replace("_", "-"),
+                    "title": data.pop("title"),
```

Comment:
And just do this here:
```suggestion
                    "title": title,
```